### ISSUE TITLE
Implement Account Challenges

### ISSUE THREAD

**Issue by dannie9099** [JRoy/InstagramLive-PHP#14]
Please I have been getting this error 

Loading InstagramLive-PHP v0.4...
Logging into Instagram...
Account Flagged: Please sign out of all phones and try logging into instagram.com from this computer before trying to run this script again!

I have signed out of phones. and signed in using my computer yet error persists...

Kindly help out...

Thank you

**Comment by andrewv** [comment]
I'm having the same issue - I allow the security requests online but keep having it..

**Comment by JRoy** [comment]
I am aware of this issue happening in some cases and hope on adding a possible fix to this in v0.t

**Comment by mrkrnfl** [comment]
Hi, i'm getting this error. Can you help about it? 
```
C:\Users\Emre\Desktop\InstagramLive-PHP>php goLive.php
Loading InstagramLive-PHP v0.4...
Logging into Instagram...
Error While Logging in to Instagram: Network: CURL error 60: SSL certificate problem: unable to get local issuer certificate (see http://curl.haxx.se/libcurl/c/libcurl-errors.html).

```

**Comment by JRoy** [comment]
@mrkrnfl: https://stackoverflow.com/a/34883260

**Comment by JRoy** [comment]
Please ignore, for future reference:
```php
<?php

require __DIR__ . '/vendor/autoload.php';

//Enter these options
$username            = '';
$password            = '';
$verification_method = 0; 	//0 = SMS, 1 = Email


class ExtendedInstagram extends Instagram {
	public function changeUser( $username, $password ) {
		$this->_setUser( $username, $password );
	}
}

function readln( $prompt ) {
	if ( PHP_OS === 'WINNT' ) {
		echo "$prompt ";

		return trim( (string) stream_get_line( STDIN, 6, "\n" ) );
	}

	return trim( (string) readline( "$prompt " ) );
}

$ig = new ExtendedInstagram();

try {
	$loginResponse = $ig->login( $username, $password );
	echo 'Logged in!';
} catch ( Exception $exception ) {

	$response = $exception->getResponse();

	if ($exception instanceof ChallengeRequiredException
	&& $response->getErrorType() === 'checkpoint_challenge_required') {

		sleep(3);

		$checkApiPath = substr( $response->getChallenge()->getApiPath(), 1);
		$customResponse = $ig->request($checkApiPath)
									->setNeedsAuth(false)
									->addPost('choice', $verification_method)
									->addPost('_uuid', $ig->uuid)
									->addPost('guid', $ig->uuid)
									->addPost('device_id', $ig->device_id)
									->addPost('_uid', $ig->account_id)
									->addPost('_csrftoken', $ig->client->getToken())
									->getDecodedResponse();

	} else {
		echo "Not a challenge required exception...\n";
		exit;
	}

	try {

		if ($customResponse['status'] === 'ok' && $customResponse['action'] === 'close') {
			echo 'Checkpoint bypassed';
                        exit();
		}

		$code = readln( 'Code that you received via ' . ( $verification_method ? 'email' : 'sms' ) . ':' );
		$ig->changeUser( $username, $password );
		$customResponse = $ig->request($checkApiPath)
									->setNeedsAuth(false)
									->addPost('security_code', $code)
									->addPost('_uuid', $ig->uuid)
									->addPost('guid', $ig->uuid)
									->addPost('device_id', $ig->device_id)
									->addPost('_uid', $ig->account_id)
									->addPost('_csrftoken', $ig->client->getToken())
									->getDecodedResponse();

		if ($customResponse['status'] === 'ok' && (int) $customResponse['logged_in_user']['pk'] === (int) $user_id ) {
			echo 'Finished, logged in successfully! Run this file again to validate that it works.';
		} else {
			echo "Probably finished...\n";
			var_dump( $customResponse );
		}
	} catch ( Exception $ex ) {
		echo $ex->getMessage();
	}
}
```

**Comment by JRoy** [comment]
@shuggaexo, This will be coming in 0.6, I am just yet to actually finish it. Expect a beta soon for it tho

**Comment by JRoy** [comment]
Hey guys watching this thread: I've added this and you guys should check out #27 to try the beta :)

**Comment by andrewv** [comment]
Amazing, will do! Thanks!

**Comment by andrewv** [comment]
Hey @JRoy - it worked! We verified our account via email and then logged in successfully! Thank you SO much - this is amazing! Great work!

**Comment by JRoy** [comment]
Whoops, forgot to close this but this feature has been fully implemented as of 0.6 :)
